server/src/domain/storage-template/storage-template.service.spec.ts

```diff
@@ -236,12 +271,47 @@ describe(StorageTemplateService.name, () => {
       });
     });
 
+    it('should not update the database if the move fails due to incorrect newPath filesize', async () => {
+      assetMock.getAll.mockResolvedValue({
+        items: [assetStub.image],
+        hasNextPage: false,
+      });
+      storageMock.rename.mockRejectedValue({ code: 'EXDEV' });
+      userMock.getList.mockResolvedValue([userStub.user1]);
+      moveMock.create.mockResolvedValue({
+        id: '123',
+        entityId: assetStub.image.id,
+        pathType: AssetPathType.ORIGINAL,
+        oldPath: assetStub.image.originalPath,
+        newPath: 'upload/library/user-id/2023/2023-02-23/asset-id.jpg',
+      });
+      // @ts-expect-error we only use the size property, no point implementing all the other properties
+      when(storageMock.stat).calledWith('upload/library/user-id/2023/2023-02-23/asset-id.jpg').mockResolvedValue({
+        size: 100,
+      });
```

Comment:
I would prefer `.mockResolvedValue({ size: 100 } as Stats)` here instead of having this annotation

Reply:
Ah yes, forgot you can do that